src/geogenalg/application/generalize_clusters_to_centroids.py

```diff
@@ -15,31 +15,44 @@
 from pandas import Series, concat
 from shapely import Point, Polygon, box
 
-from geogenalg.application import BaseAlgorithm
+from geogenalg.application import BaseAlgorithm, supports_identity
 from geogenalg.cluster import get_cluster_centroids
 from geogenalg.core.exceptions import GeometryTypeError
 from geogenalg.core.geometry import mean_z
 from geogenalg.utility.validation import check_gdf_geometry_type
 
 
+@supports_identity
 @dataclass(frozen=True)
 class GeneralizePointClustersAndPolygonsToCentroids(BaseAlgorithm):
-    """Reduces polygons and point clusters to single points."""
+    """Reduces polygons and point clusters to single points.
+
+    Input can be point and/or polygons for which centroids are possibly generated.
+
+    Reference data can be polygons, within which the centroids will be placed.
+
+    Output is generalized data, which contains the newly
+    created centroids and all input features which were not turned into
+    centroids. These can be distinguished by the value in the column of
+    the name set by the feature_type_column attribute.
+    """
 
-    """Points within this distance of each other will be clustered."""
     cluster_distance: float
-    """Polygons with area smaller than this will be turned into a point."""
+    """Points within this distance of each other will be clustered."""
     polygon_min_area: float
-    """Name of column containing a unique identifier of input features."""
+    """Polygons with area smaller than this will be turned into a point."""
     unique_id_column: str
-    """Name of column containing type of output feature."""
+    """Name of column containing a unique identifier of input features."""
     feature_type_column: str
+    """Name of column containing type of output feature."""
+    aggregation_functions: dict[str, Callable[[Series], Any] | str] | None
     """Dictionary containing keys corresponding to a column in a GeoDataFrame
     and a function which will aggregate the column's values when creating
     centroid from multiple points. If the function is given as a string, it
     must correspond to Pandas's aggregation function names. If no function is
     given "first" will be used by default."""
-    aggregation_functions: dict[str, Callable[[Series], Any] | str] | None
+    reference_key: str = "mask"
+    "Reference data key to use as a mask layer for centroid placement."
```

Comment:
For this algorithm use case a single reference key is proably enough for now?

In the future we migth need to check this api, whether its better to pass in multiple reference keys and the algorithm concatenates all inputs if necessary (easier for integrating the alg, since we might need to pass for example `lake_part,sea_part,watercourse_part_area` as reference), or pass in a single key and we add the concatenation on the integration side (easier for the alg).

In both cases its still better for the key to be configurable (with default).